### ISSUE TITLE
allowing custom message signature

### ISSUE THREAD

**Issue by harpagon210** [ledgerconnect/hivesigner#8]
Adding the ability to sign custom messages would be a very good add-on, here's what I think the custom message feature should allow:

- choose the key required to sign the message (posting or active)
- allow signing a SHA256 payload (maybe allow signing a non SHA256 message that would be hashed during the signing process?)
- allow calling a callback url
- the callback payload would contain the account and the signature (and the original payload? )

A signing url could look like:
https://hivesigner.com/sign/custom-message? account=harpagon&key=posting&payload=9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08&cb=https://my-url.com/callback

This feature would help users to use the Hive ecosystem without any additional browser extension and without having to trust any website asking for a signature.

**Comment by krischik** [comment]
Yes. Browser extensions suck. If you use the Hive ecosystem then you probably also have a steem account.

And with hivesigner/ steemconnect and steemkeychain/hivekeychain competing against each other you end up with 4 of those cluttering up your browser.

So any feature which makes is easier to use without a browser extension is more than welcome.

Which is also the reason why my current favourite is the underdog peaklock.

**Comment by feruzm** [comment]
@harpagon210 Do you have example of this being/can be used anywhere?

**Comment by harpagon210** [comment]
@feruzm this could be used for basically anything, I personally need it to allow users to log into my app without having to install any additional tool, they can just click on a link, enter their credentials and that's it, the callback tells me if the signature process was successful or not.

**Comment by feruzm** [comment]
@harpagon210 There were few other request in last couple month about this. I will see what we can do, feel free to suggest PR or improvements. Right now you can use `scope: login` just like 6. Other use cases suggested here: https://hivesigner.com/developers, which allows user to login without giving posting authority to your app account.
